src/instructlab/model/backends/llama_cpp.py

```diff
@@ -89,19 +90,37 @@ def create_server_process(self, port: int) -> multiprocessing.Process:
     def run_detached(
         self, http_client: httpx.Client | None = None, background: bool = True
     ) -> str:
+        if check_api_base(self.api_base, http_client):
+            return self.api_base
         try:
-            llama_cpp_server_process, _, api_base = ensure_server(
-                backend=LLAMA_CPP,
-                api_base=self.api_base,
-                http_client=http_client,
-                host=self.host,
-                port=self.port,
-                queue=self.queue,
-                server_process_func=self.create_server_process,
-            )
-            self.process = llama_cpp_server_process or self.process
-            self.api_base = api_base or self.api_base
+            self.port = free_tcp_ipv4_port(self.host)
+            # start new server
+            self.api_base = str(get_api_base(f"{self.host}:{self.port}"))
+            logger.debug(f"Starting a temporary server at {self.api_base}...")
+            self.process = self.create_server_process(self.port)
+            self.process.start()
+
+            # in case the server takes some time to fail we wait a bit
+            logger.debug("Waiting for the server to start...")
+            count = 0
+            while self.process.is_alive():
+                sleep(0.1)
```

Comment:
I realize it's not new code, but why is this sleep so short?

Reply:
Usually when I see this it's to prevent a busy loop, or to ensure the process yields to others each time around instead of just eating up the CPU. In this case, it would be trying to connect to the API server as fast as possible for 50 tries without it. Now it's doing it 10 times a second (a little more than a second in practice).

That's still more aggressive than necessary, though, IMO.

Reply:
I was curious so I dug in a little further.  The list_models call can take a few seconds respond while the server is coming up.

Examples printing time.time() in the loop:
```
1720913810.889125
1720913813.3796103
```

```
1720913780.4392679
1720913783.166862
```

So the sleep isn't really doing the heavy lifting of the waiting in this case.  Makes sense why it was made so short.


@JamesKunstle ^^ related to a conversation we were having around vllm